diff --git a/runtime/Makefile b/runtime/Makefile
index 744955c0d..44ff134f4 100644
--- a/runtime/Makefile
+++ b/runtime/Makefile
@@ -273,7 +273,9 @@ caml/version.h : $(ROOTDIR)/tools/make-version-header.sh $(ROOTDIR)/VERSION
 # Libraries and programs
 
 ocamlrun$(EXE): prims.$(O) libcamlrun.$(A)
-	$(call MAKE_OCAMLRUN,$@,$^ $(LIBS))
+	# $(call MAKE_OCAMLRUN,$@,$^ $(LIBS))
+	cp $(OCAMLRUN) $@
+	chmod 755 $@
 
 libcamlrun.$(A): $(libcamlrun_OBJECTS)
 	$(call MKLIB,$@, $^)
diff --git a/tools/Makefile b/tools/Makefile
index dbad0b74f..2cb607c52 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -352,9 +352,8 @@ CAMLTEX= $(ROOTDIR)/compilerlibs/ocamlcommon.cma \
 
 caml-tex: INCLUDES += $(addprefix -I $(ROOTDIR)/otherlibs/,str $(UNIXLIB))
 caml-tex: $(CAMLTEX)
-	$(ROOTDIR)/runtime/ocamlrun $(ROOTDIR)/ocamlc -nostdlib \
-	                            -I $(ROOTDIR)/stdlib $(LINKFLAGS) -linkall \
-	                            -o $@ -no-alias-deps $(CAMLTEX)
+	$(CAMLC) -I $(ROOTDIR)/stdlib $(LINKFLAGS) -linkall \
+	         -o $@ -no-alias-deps $(CAMLTEX)
 
 # we need str and unix which depend on the bytecode version of other tools
 # thus we delay building caml-tex to the opt.opt stage
