 scripts/install.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/install.js b/scripts/install.js
index 1c544d551..fc18069c6 100644
--- a/scripts/install.js
+++ b/scripts/install.js
@@ -41,6 +41,9 @@ function provideNinja() {
       cwd: ninja_source_dir,
       stdio: [0, 1, 2]
     });
+    cp.execSync('patch -p1 -i ninja.patch', {
+      stdio: [0, 1, 2]
+    });
     console.log("No prebuilt Ninja, building Ninja now");
     var build_ninja_command = "./configure.py --bootstrap";
     cp.execSync(build_ninja_command, {
