diff --git a/src/frontend/ocamlmerlin/dune b/src/frontend/ocamlmerlin/dune
index b4f149e3..6f3c2b5a 100644
--- a/src/frontend/ocamlmerlin/dune
+++ b/src/frontend/ocamlmerlin/dune
@@ -6,7 +6,7 @@
  (public_name ocamlmerlin-server)
  (flags -open Merlin_utils -open Merlin_analysis)
  (modules (:standard \ gen_ccflags))
- (libraries config yojson merlin_analysis merlin_kernel
+ (libraries config yojson merlin_analysis merlin_kernel camlp-streams
             merlin_utils os_ipc parsing query_protocol query_commands
             typing utils))
 
