diff --git a/src/heap/dune b/src/heap/dune
index 63631b77e..d2c526519 100644
--- a/src/heap/dune
+++ b/src/heap/dune
@@ -24,8 +24,7 @@
   (wrapped false)
   (modules
     prefix
-    sharedMem
-    value)
+    sharedMem)
   (preprocess (pps ppx_deriving.std visitors.ppx))
   (libraries
     heap_libc
diff --git a/src/parser/dune b/src/parser/dune
index add19f320..91290b5f5 100644
--- a/src/parser/dune
+++ b/src/parser/dune
@@ -1,6 +1,8 @@
 (library
   (name flow_parser)
+  (public_name flow_parser)
   (wrapped false)
+  (flags -w -16)
   (modules (:standard \ flow_parser_js flow_parser_dot_js))
   (libraries
     sedlex
