diff --git a/src/reason-parser/reason_syntax_util.ml b/src/reason-parser/reason_syntax_util.ml
index 05f01ce9..5cc4356d 100644
--- a/src/reason-parser/reason_syntax_util.ml
+++ b/src/reason-parser/reason_syntax_util.ml
@@ -17,7 +17,7 @@
 open Ppxlib
 
 (* Rename labels in function definition/application and records *)
-let rename_labels = ref false
+let rename_labels = ref true
 
 (** Check to see if the string `s` is made up of `keyword` and zero or more
     trailing `_` characters. *)
