commit eaf7157bfb33d6f14ab33685072ad49668896c57
Author: Antonio Nuno Monteiro <anmonteiro@gmail.com>
Date:   Thu Apr 21 12:32:48 2022 -0700

    Fix for 5.00

diff --git a/core_thread/src/pthread_np_stubs.c b/core_thread/src/pthread_np_stubs.c
index 8d036fb..d4cfe82 100644
--- a/core_thread/src/pthread_np_stubs.c
+++ b/core_thread/src/pthread_np_stubs.c
@@ -66,4 +66,6 @@ CAMLprim value pthread_np_getaffinity_self()
   }
   CAMLreturn(v_cpus);
 }
+#else
+void unused(){};
 #endif /* JSC_PTHREAD_NP */
diff --git a/core_unix/src/core_unix_stubs.c b/core_unix/src/core_unix_stubs.c
index 3172803..c6f25bc 100644
--- a/core_unix/src/core_unix_stubs.c
+++ b/core_unix/src/core_unix_stubs.c
@@ -1282,7 +1282,7 @@ CAMLprim value core_unix_sched_setscheduler(
 #warning "_POSIX_PRIORITY_SCHEDULING not present; sched_setscheduler undefined"
 CAMLprim value core_unix_sched_setscheduler(
   value __unused v_pid, value __unused v_policy, value __unused v_priority)
-{  invalid_argument("sched_setscheduler unimplemented"); }
+{  caml_invalid_argument("sched_setscheduler unimplemented"); }
 #endif
 
 
diff --git a/linux_ext/src/linux_ext_stubs.c b/linux_ext/src/linux_ext_stubs.c
index a9983e8..c840aa6 100644
--- a/linux_ext/src/linux_ext_stubs.c
+++ b/linux_ext/src/linux_ext_stubs.c
@@ -800,4 +800,5 @@ CAMLprim value core_linux_setxattr(value v_path, value v_name, value v_value, va
 
 typedef int avoid_empty_translation_unit_compilation_error;
 
+void unused(){};
 #endif /* JSC_LINUX_EXT */
diff --git a/unix_pseudo_terminal/src/unix_pseudo_terminal_stubs.c b/unix_pseudo_terminal/src/unix_pseudo_terminal_stubs.c
index cb8beb2..b017674 100644
--- a/unix_pseudo_terminal/src/unix_pseudo_terminal_stubs.c
+++ b/unix_pseudo_terminal/src/unix_pseudo_terminal_stubs.c
@@ -92,4 +92,6 @@ CAMLprim value unix_ptsname(value mlfd)
   CAMLreturn (mlname);
 }
 
+#else
+void unused(){}
 #endif /* JSC_UNIX_PTY */
