diff --git a/ocaml/configure.ac b/ocaml/configure.ac
index d1224de9b..f96dd706d 100644
--- a/ocaml/configure.ac
+++ b/ocaml/configure.ac
@@ -2125,7 +2125,18 @@ AS_CASE([$host],
   [*-pc-windows],
     [PTHREAD_LIBS="-l:libpthread.lib"],
   [*-apple-darwin*],
-    [PTHREAD_LIBS="-pthread"],
+    [PTHREAD_LIBS="-pthread"
+     AX_PTHREAD(
+       [common_cflags="$common_cflags $PTHREAD_CFLAGS"
+       saved_CFLAGS="$CFLAGS"
+       saved_LIBS="$LIBS"
+       CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+       LIBS="$LIBS $PTHREAD_LIBS"
+       AC_CHECK_FUNC([sigwait], [AC_DEFINE([HAS_SIGWAIT])])
+       LIBS="$saved_LIBS"
+       CFLAGS="$saved_CFLAGS"],
+       [AC_MSG_ERROR(m4_normalize([POSIX threads are required but not supported on
+         this platform]))])],
   [AX_PTHREAD(
     [common_cflags="$common_cflags $PTHREAD_CFLAGS"
     saved_CFLAGS="$CFLAGS"
diff --git a/ocaml/parsing/dune b/ocaml/parsing/dune
index 6933891b2..79d9cf60f 100644
--- a/ocaml/parsing/dune
+++ b/ocaml/parsing/dune
@@ -18,14 +18,22 @@
   (with-stdout-to %{targets}
    ;; Partial applications of the form Obj.magic f x in menhirLib cause
    ;; an issue with locals, so rewrite these to Obj.magic (f x)
-   (bash "cat $(menhir --suggest-menhirLib)/menhirLib.ml |
+   (bash "menhir_file_path=\"$(menhir --suggest-menhirLib 2>/dev/null)/menhirLib.ml\"
+          if [ ! -f \"$menhir_file_path\" ]; then
+              menhir_file_path=\"$(ocamlfind query menhirLib)/menhirLib.ml\"
+          fi
+          cat \"$menhir_file_path\" |
           sed 's/\\b\\(in\\|then\\|with\\|else\\)\\b/@@@\\1/g; s/Obj.magic \\([a-z0-9_]\\+\\( [a-z0-9_]\\+\\)\\+\\)/Obj.magic (\\1)/g;  s/@@@//g'"))))
 
 (rule
  (targets camlinternalMenhirLib.mli)
  (action
   (with-stdout-to %{targets}
-   (bash "cat $(menhir --suggest-menhirLib)/menhirLib.mli"))))
+   (bash "menhir_file_path=\"$(menhir --suggest-menhirLib 2>/dev/null)/menhirLib.mli\"
+          if [ ! -f \"$menhir_file_path\" ]; then
+              menhir_file_path=\"$(ocamlfind query menhirLib)/menhirLib.mli\"
+          fi
+          cat \"$menhir_file_path\""))))
 
 (ocamllex
  (modules lexer)
