From 2059959a751c0dd7e65fc7e001cf5dc2352da90b Mon Sep 17 00:00:00 2001
From: Thomas Gazagnaire <thomas@gazagnaire.org>
Date: Thu, 28 Apr 2022 14:48:23 +0200
Subject: [PATCH] Fix compilation with OCaml 5

---
 src/trace.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/trace.ml b/src/trace.ml
index aaee495..e3e4b7d 100644
--- a/src/trace.ml
+++ b/src/trace.ml
@@ -348,7 +348,7 @@ let make_writer dest ?getpid (info : Info.t) =
 
 module IntTbl = Hashtbl.MakeSeeded (struct
   type t = int
-  let hash _seed (id : t) =
+  let seeded_hash _seed (id : t) =
     let h = id * 189696287 in
     h lxor (h lsr 23)
   let equal (a : t) (b : t) = a = b
